Normalizer/Flat/ReferenceDataNormalizer.php

```diff
@@ -0,0 +1,131 @@
+use Pim\Component\ReferenceData\Model\ReferenceDataInterface;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * @author    Romain Monceau <romain@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ */
+class ReferenceDataNormalizer implements NormalizerInterface
+{
+    /** @var TargetEntityResolver */
+    protected $targetEntityResolver;
+
+    /** @var ClassMetadataRegistry */
+    protected $classMetadataRegistry;
+
+    /** @var PropertyAccessorInterface */
+    protected $propertyAccessor;
+
+    /** @var NormalizerInterface */
+    protected $transNormalizer;
+
+    /** @var string[] */
+    protected $skippedFields = [];
+
+    /**
+     * @param TargetEntityResolver $targetEntityResolver
+     * @param ClassMetadataRegistry $classMetadataRegistry
+     * @param PropertyAccessorInterface $propertyAccessor
+     * @param NormalizerInterface $transNormalizer
+     */
+    public function __construct(
+        TargetEntityResolver $targetEntityResolver,
+        ClassMetadataRegistry $classMetadataRegistry,
+        PropertyAccessorInterface $propertyAccessor,
+        NormalizerInterface $transNormalizer
+    ) {
+        $this->targetEntityResolver = $targetEntityResolver;
+        $this->classMetadataRegistry = $classMetadataRegistry;
+        $this->propertyAccessor = $propertyAccessor;
+        $this->transNormalizer = $transNormalizer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($object, $format = null, array $context = []): array
+    {
+        $csvData = [];
+
+        $properties = $this->classMetadataRegistry->getReadableProperties($object);
+        $properties = array_diff($properties, $this->skippedFields);
+
+        foreach ($properties as $property) {
+            $propertyValue = $this->propertyAccessor->getValue($object, $property);
+
+            if (is_object($propertyValue)) {
+                $normalizedData = $this->normalizeLinkedObject($object, $property, $propertyValue, $format, $context);
+                if (is_array($normalizedData)) {
+                    $csvData = array_merge($csvData, $normalizedData);
+                }
+            } else {
+                if (!is_array($propertyValue)) {
+                    $csvData[$property] = $propertyValue;
+                }
+            }
+        }
+
+        return $csvData;
+    }
+
+    /**
+     * Normalized linked object property
+     *
+     * @param mixed $object
+     * @param string $property
+     * @param string $format
+     * @param array $context
+     *
+     * @return array|null
+     */
+    protected function normalizeLinkedObject($object, $property, $propertyValue, $format, $context)
+    {
+        $targetEntityClass = $this->targetEntityResolver->getTargetEntityClass($object, $property);
+        $targetReflectionClass = $this->classMetadataRegistry->get($targetEntityClass);
```

Comment:
you should check if `$targetEntityClass` is null before doing that

Reply:
Yep good catch. Done on the resolver side, it throws an exception if nothing found